Saving a specific dashboard times out on cloud instance

### Describe the bug

Cloud customer's dashboard fails to save after a while with "This connection has been closed". We also found deadlocks on updating view count and last_viewed_at on dashboards in this instance so they could possibly be related. See logs attached

### To Reproduce

Happening on cloud, couldn't get a local repro yet, will update this issue if I manage to do it

### Expected behavior

Should save the dashboard normally

### Logs

```
{"lvl":"ERROR","lgr":"metabase.server.middleware.log","m":"PUT /api/dashboard/119 500 311877ms (10 DB calls) {:metabase-user-id 12} \n{:via\n [{:type clojure.lang.ExceptionInfo,\n   :message \"This connection has been closed.\",\n   :data\n   {:toucan2/context-trace\n    [[\"resolve connection\" {:toucan2.connection/connectable metabase.db.connection.ApplicationDB}]\n     [\"resolve connection\" {:toucan2.connection/connectable :default}]\n     [\"resolve connection\" {:toucan2.connection/connectable nil}]]},\n   :at [org.postgresql.jdbc.PgConnection checkClosed \"PgConnection.java\" 1009]}\n  {:type org.postgresql.util.PSQLException,\n   :message \"This connection has been closed.\",\n   :at [org.postgresql.jdbc.PgConnection checkClosed \"PgConnection.java\" 1009]}],\n :trace\n [[org.postgresql.jdbc.PgConnection checkClosed \"PgConnection.java\" 1009]\n  [org.postgresql.jdbc.PgConnection setAutoCommit \"PgConnection.java\" 938]\n  [com.mchange.v2.c3p0.impl.NewProxyConnection setAutoCommit \"NewProxyConnection.java\" 1401]\n  [metabase.db.connection$do_transaction invokeStatic \"connection.clj\" 161]\n  [metabase.db.connection$do_transaction invoke \"connection.clj\" 136]\n  [metabase.db.connection$do_with_transaction_primary_method_java_sql_Connection invokeStatic \"connection.clj\" 196]\n  [metabase.db.connection$do_with_transaction_primary_method_java_sql_Connection invoke \"connection.clj\" 169]\n  [methodical.impl.combo.common$partial_STAR_$fn__19889 invoke \"common.clj\" 13]\n  [methodical.util.FnWithMeta invoke \"util.clj\" 46]\n  [toucan2.connection$do_with_transaction_around_method_toucan2_connection_default invokeStatic \"connection.clj\" 249]\n  [toucan2.connection$do_with_transaction_around_method_toucan2_connection_default invoke \"connection.clj\" 245]\n  [methodical.impl.combo.common$partial_STAR_$fn__19889 invoke \"common.clj\" 13]\n  [methodical.util.FnWithMeta invoke \"util.clj\" 46]\n  [methodical.impl.standard$invoke_multifn invokeStatic \"standard.clj\" 58]\n  [methodical.impl.standard$invoke_multifn invoke \"standard.clj\" 47]\n  [methodical.impl.standard.StandardMultiFn invoke \"standard.clj\" 212]\n  [metabase.api.dashboard$update_dashboard$with_connection_STAR___105331 invoke \"dashboard.clj\" 873]\n  [toucan2.connection$bind_current_connectable_fn$fn__23214 invoke \"connection.clj\" 104]\n  [toucan2.connection$bind_current_connectable_fn$fn__23214 invoke \"connection.clj\" 104]\n  [toucan2.connection$bind_current_connectable_fn$fn__23214 invoke \"connection.clj\" 104]\n  [toucan2.jdbc.connection$do_with_connection_primary_method_javax_sql_DataSource invokeStatic \"connection.clj\" 18]\n  [toucan2.jdbc.connection$do_with_connection_primary_method_javax_sql_DataSource invoke \"connection.clj\" 15]\n  [methodical.impl.combo.common$partial_STAR_$fn__19889 invoke \"common.clj\" 12]\n  [methodical.util.FnWithMeta invoke \"util.clj\" 46]\n  [toucan2.connection$do_with_connection_around_method_toucan2_connection_default invokeStatic \"connection.clj\" 118]\n  [toucan2.connection$do_with_connection_around_method_toucan2_connection_default invoke \"connection.clj\" 106]\n  [methodical.impl.combo.common$partial_STAR_$fn__19889 invoke \"common.clj\" 12]\n  [methodical.util.FnWithMeta invoke \"util.clj\" 46]\n  [methodical.impl.standard$invoke_multifn invokeStatic \"standard.clj\" 55]\n  [methodical.impl.standard$invoke_multifn invoke \"standard.clj\" 47]\n  [methodical.impl.standard.StandardMultiFn invoke \"standard.clj\" 210]\n  [metabase.db.connection$do_with_connection_primary_method_default invokeStatic \"connection.clj\" 132]\n  [metabase.db.connection$do_with_connection_primary_method_default invoke \"connection.clj\" 130]\n  [methodical.impl.combo.common$partial_STAR_$fn__19889 invoke \"common.clj\" 12]\n  [methodical.util.FnWithMeta invoke \"util.clj\" 46]\n  [toucan2.connection$do_with_connection_around_method_toucan2_connection_default invokeStatic \"connection.clj\" 118]\n  [toucan2.connection$do_with_connection_around_method_toucan2_connection_default invoke \"connection.clj\" 106]\n  [methodical.impl.combo.common$partial_STAR_$fn__19889 invoke \"common.clj\" 12]\n  [methodical.util.FnWithMeta invoke \"util.clj\" 46]\n  [methodical.impl.standard$invoke_multifn invokeStatic \"standard.clj\" 55]\n  [methodical.impl.standard$invoke_multifn invoke \"standard.clj\" 47]\n  [methodical.impl.standard.StandardMultiFn invoke \"standard.clj\" 210]\n  [toucan2.connection$do_with_connection_primary_method_ invokeStatic \"connection.clj\" 204]\n  [toucan2.connection$do_with_connection_primary_method_ invoke \"connection.clj\" 194]\n  [methodical.impl.combo.common$partial_STAR_$fn__19889 invoke \"common.clj\" 12]\n  [methodical.util.FnWithMeta invoke \"util.clj\" 46]\n  [toucan2.connection$do_with_connection_around_method_toucan2_connection_default invokeStatic \"connection.clj\" 118]\n  [toucan2.connection$do_with_connection_around_method_toucan2_connection_default invoke \"connection.clj\" 106]\n  [methodical.impl.combo.common$partial_STAR_$fn__19889 invoke \"common.clj\" 12]\n  [methodical.util.FnWithMeta invoke \"util.clj\" 46]\n  [methodical.impl.standard$invoke_multifn invokeStatic \"standard.clj\" 55]\n  [methodical.impl.standard$invoke_multifn invoke \"standard.clj\" 47]\n  [methodical.impl.standard.StandardMultiFn invoke \"standard.clj\" 210]\n  [metabase.api.dashboard$update_dashboard invokeStatic \"dashboard.clj\" 873]\n  [metabase.api.dashboard$update_dashboard invoke \"dashboard.clj\" 849]\n  [metabase.api.dashboard$fn__105364$fn__105371 invoke \"dashboard.clj\" 973]\n  [metabase.api.macros$_core_fn96998__96999$core_fn__97000 invoke \"macros.clj\" 415]\n  [metabase.api.macros$endpoint_handler_STAR_97042__97044$handler__97048 invoke \"macros.clj\" 611]\n  [metabase.api.macros$build_ns_handler97075__97076$ns_handler_STAR___97077 invoke \"macros.clj\" 675]\n  [metabase.api.open_api.HandlerWithOpenAPISpec invoke \"open_api.clj\" 42]\n  [metabase.server.middleware.auth$enforce_authentication$fn__107885 invoke \"auth.clj\" 23]\n  [metabase.api.open_api.HandlerWithOpenAPISpec invoke \"open_api.clj\" 42]\n  [metabase.api.util.handlers$_route_map_handler$fn__107841 invoke \"handlers.clj\" 23]\n  [metabase.api.open_api.HandlerWithOpenAPISpec invoke \"open_api.clj\" 42]\n  [compojure.core$routes$fn__107778$f__107779 invoke \"core.clj\" 198]\n  [compojure.core$routes$fn__107778$f__107779$respond_SINGLEQUOTE___107780 invoke \"core.clj\" 197]\n  [metabase.api.util.handlers$_route_map_handler$fn__107841 invoke \"handlers.clj\" 24]\n  [metabase.api.open_api.HandlerWithOpenAPISpec invoke \"open_api.clj\" 42]\n  [clojure.lang.Var invoke \"Var.java\" 395]\n  [compojure.core$routes$fn__107778$f__107779 invoke \"core.clj\" 198]\n  [compojure.core$routes$fn__107778 invoke \"core.clj\" 200]\n  [metabase.api.open_api.HandlerWithOpenAPISpec invoke \"open_api.clj\" 42]\n  [metabase.server.routes$fn__118198$fn__118199 invoke \"routes.clj\" 70]\n  [compojure.core$routes$fn__107778$f__107779 invoke \"core.clj\" 198]\n  [compojure.core$routes$fn__107778 invoke \"core.clj\" 200]\n  [compojure.core$make_context$handler__107806 invoke \"core.clj\" 290]\n  [compojure.core$make_context$fn__107810 invoke \"core.clj\" 300]\n  [compojure.core$routes$fn__107778$f__107779 invoke \"core.clj\" 198]\n  [compojure.core$routes$fn__107778$f__107779$respond_SINGLEQUOTE___107780 invoke \"core.clj\" 197]\n  [compojure.core$wrap_route_matches$fn__107759 invoke \"core.clj\" 153]\n  [compojure.core$routes$fn__107778$f__107779 invoke \"core.clj\" 198]\n  [compojure.core$routes$fn__107778$f__107779$respond_SINGLEQUOTE___107780 invoke \"core.clj\" 197]\n  [compojure.core$wrap_route_matches$fn__107759 invoke \"core.clj\" 153]\n  [compojure.core$routes$fn__107778$f__107779 invoke \"core.clj\" 198]\n  [compojure.core$routes$fn__107778$f__107779$respond_SINGLEQUOTE___107780 invoke \"core.clj\" 197]\n  [compojure.core$wrap_route_matches$fn__107759 invoke \"core.clj\" 153]\n  [compojure.core$routes$fn__107778$f__107779 invoke \"core.clj\" 198]\n  [compojure.core$routes$fn__107778$f__107779$respond_SINGLEQUOTE___107780 invoke \"core.clj\" 197]\n  [compojure.core$wrap_route_matches$fn__107759 invoke \"core.clj\" 153]\n  [compojure.core$routes$fn__107778$f__107779 invoke \"core.clj\" 198]\n  [compojure.core$routes$fn__107778$f__107779$respond_SINGLEQUOTE___107780 invoke \"core.clj\" 197]\n  [metabase.api.util.handlers$_route_map_handler$fn__107841 invoke \"handlers.clj\" 24]\n  [metabase.api.open_api.HandlerWithOpenAPISpec invoke \"open_api.clj\" 42]\n  [metabase.api.util.handlers$_route_map_handler$fn__107841 invoke \"handlers.clj\" 23]\n  [metabase.api.open_api.HandlerWithOpenAPISpec invoke \"open_api.clj\" 42]\n  [clojure.lang.Var invoke \"Var.java\" 395]\n  [compojure.core$routes$fn__107778$f__107779 invoke \"core.clj\" 198]\n  [compojure.core$routes$fn__107778 invoke \"core.clj\" 200]\n  [metabase.server.middleware.exceptions$catch_uncaught_exceptions$fn__110095 invoke \"exceptions.clj\" 111]\n  [metabase.server.middleware.exceptions$catch_api_exceptions$fn__110092 invoke \"exceptions.clj\" 100]\n  [metabase.server.middleware.log$log_api_call$fn__124557$fn__124558$fn__124559 invoke \"log.clj\" 232]\n  [metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info invokeStatic \"diagnostic.clj\" 17]\n  [metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info invoke \"diagnostic.clj\" 12]\n  [metabase.server.middleware.log$log_api_call$fn__124557$fn__124558 invoke \"log.clj\" 223]\n  [toucan2.execute$do_with_call_counts invokeStatic \"execute.clj\" 112]\n  [toucan2.execute$do_with_call_counts invoke \"execute.clj\" 103]\n  [metabase.server.middleware.log$log_api_call$fn__124557 invoke \"log.clj\" 222]\n  [metabase.server.middleware.browser_cookie$ensure_browser_id_cookie$fn__133966 invoke \"browser_cookie.clj\" 40]\n  [metabase.server.middleware.security$add_security_headers$fn__110051 invoke \"security.clj\" 275]\n  [metabase.server.middleware.json$wrap_json_body$fn__50836 invoke \"json.clj\" 128]\n  [metabase.server.middleware.offset_paging$handle_paging$fn__132578 invoke \"offset_paging.clj\" 48]\n  [metabase.server.middleware.json$wrap_streamed_json_response$fn__50824 invoke \"json.clj\" 83]\n  [ring.middleware.keyword_params$wrap_keyword_params$fn__134233 invoke \"keyword_params.clj\" 55]\n  [ring.middleware.params$wrap_params$fn__134252 invoke \"params.clj\" 77]\n  [metabase.server.middleware.misc$maybe_set_site_url$fn__123924 invoke \"misc.clj\" 59]\n  [metabase.server.middleware.session$reset_session_timeout$fn__126992 invoke \"session.clj\" 280]\n  [metabase.server.middleware.session$bind_current_user$fn__126985$fn__126986 invoke \"session.clj\" 255]\n  [metabase.request.session$do_with_current_user invokeStatic \"session.clj\" 47]\n  [metabase.request.session$do_with_current_user invoke \"session.clj\" 30]\n  [metabase.server.middleware.session$bind_current_user$fn__126985 invoke \"session.clj\" 254]\n  [metabase.server.middleware.session$wrap_current_user_info$fn__126977 invoke \"session.clj\" 214]\n  [metabase.analytics.sdk$embedding_mw$embedding_mw_fn__64207 invoke \"sdk.clj\" 79]\n  [metabase.server.middleware.session$wrap_session_key$fn__126946 invoke \"session.clj\" 84]\n  [metabase.server.middleware.auth$wrap_static_api_key$fn__107893 invoke \"auth.clj\" 38]\n  [ring.middleware.cookies$wrap_cookies$fn__134153 invoke \"cookies.clj\" 200]\n  [metabase.server.middleware.misc$add_content_type$fn__123906 invoke \"misc.clj\" 28]\n  [metabase.server.middleware.misc$disable_streaming_buffering$fn__123932 invoke \"misc.clj\" 75]\n  [ring.middleware.gzip$wrap_gzip$fn__134195 invoke \"gzip.clj\" 86]\n  [metabase.server.middleware.request_id$wrap_request_id$fn__122366 invoke \"request_id.clj\" 10]\n  [metabase.server.middleware.misc$bind_request$fn__123935$fn__123936 invoke \"misc.clj\" 87]\n  [metabase.request.current$do_with_current_request invokeStatic \"current.clj\" 19]\n  [metabase.request.current$do_with_current_request invoke \"current.clj\" 15]\n  [metabase.server.middleware.misc$bind_request$fn__123935 invoke \"misc.clj\" 86]\n  [metabase.server.middleware.ssl$redirect_to_https_middleware$fn__133982 invoke \"ssl.clj\" 51]\n  [clojure.lang.Var invoke \"Var.java\" 395]\n  [metabase.server.instance$async_proxy_handler$fn__110119 invoke \"instance.clj\" 78]\n  [metabase.server.instance.proxy$org.eclipse.jetty.ee9.servlet.ServletHandler$ff19274a doHandle nil -1]\n  [org.eclipse.jetty.ee9.nested.ScopedHandler handle \"ScopedHandler.java\" 127]\n  [metabase.server.statistics_handler$new_handler$fn__97295 invoke \"statistics_handler.clj\" 105]\n  [metabase.server.statistics_handler.proxy$org.eclipse.jetty.ee9.nested.HandlerWrapper$ff19274a handle nil -1]\n  [org.eclipse.jetty.ee9.nested.HandlerWrapper handle \"HandlerWrapper.java\" 124]\n  [org.eclipse.jetty.ee9.nested.ScopedHandler nextHandle \"ScopedHandler.java\" 197]\n  [org.eclipse.jetty.ee9.nested.ContextHandler doHandle \"ContextHandler.java\" 1045]\n  [org.eclipse.jetty.ee9.nested.ScopedHandler nextScope \"ScopedHandler.java\" 164]\n  [org.eclipse.jetty.ee9.servlet.ServletHandler doScope \"ServletHandler.java\" 483]\n  [metabase.server.instance.proxy$org.eclipse.jetty.ee9.servlet.ServletHandler$ff19274a doScope nil -1]\n  [org.eclipse.jetty.ee9.nested.ScopedHandler nextScope \"ScopedHandler.java\" 162]\n  [org.eclipse.jetty.ee9.nested.ContextHandler doScope \"ContextHandler.java\" 966]\n  [org.eclipse.jetty.ee9.nested.ScopedHandler handle \"ScopedHandler.java\" 125]\n  [org.eclipse.jetty.ee9.nested.ContextHandler handle \"ContextHandler.java\" 1719]\n  [org.eclipse.jetty.ee9.nested.HttpChannel$RequestDispatchable dispatch \"HttpChannel.java\" 1564]\n  [org.eclipse.jetty.ee9.nested.HttpChannel dispatch \"HttpChannel.java\" 723]\n  [org.eclipse.jetty.ee9.nested.HttpChannel handle \"HttpChannel.java\" 511]\n  [org.eclipse.jetty.ee9.nested.ContextHandler$CoreContextHandler$CoreToNestedHandler handle \"ContextHandler.java\" 2943]\n  [org.eclipse.jetty.server.handler.ContextHandler handle \"ContextHandler.java\" 1064]\n  [org.eclipse.jetty.server.Server handle \"Server.java\" 182]\n  [org.eclipse.jetty.server.internal.HttpChannelState$HandlerInvoker run \"HttpChannelState.java\" 662]\n  [org.eclipse.jetty.server.internal.HttpConnection onFillable \"HttpConnection.java\" 416]\n  [org.eclipse.jetty.io.AbstractConnection$ReadCallback succeeded \"AbstractConnection.java\" 322]\n  [org.eclipse.jetty.io.FillInterest fillable \"FillInterest.java\" 99]\n  [org.eclipse.jetty.io.SelectableChannelEndPoint$1 run \"SelectableChannelEndPoint.java\" 53]\n  [org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy runTask \"AdaptiveExecutionStrategy.java\" 480]\n  [org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy consumeTask \"AdaptiveExecutionStrategy.java\" 443]\n  [org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy tryProduce \"AdaptiveExecutionStrategy.java\" 293]\n  [org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy run \"AdaptiveExecutionStrategy.java\" 201]\n  [org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread run \"ReservedThreadExecutor.java\" 311]\n  [org.eclipse.jetty.util.thread.QueuedThreadPool runJob \"QueuedThreadPool.java\" 979]\n  [org.eclipse.jetty.util.thread.QueuedThreadPool$Runner doRunJob \"QueuedThreadPool.java\" 1209]\n  [org.eclipse.jetty.util.thread.QueuedThreadPool$Runner run \"QueuedThreadPool.java\" 1164]\n  [java.lang.Thread run nil -1]],\n :cause \"This connection has been closed.\",\n :message \"This connection has been closed.\"}\n"}
{"lvl":"ERROR","lgr":"metabase.events.view-log","m":"Failed to update dashboard last_viewed_at","exception":{"exception_class":"clojure.lang.ExceptionInfo","exception_message":"This connection has been closed.","stacktrace":"clojure.lang.ExceptionInfo: This connection has been closed. {:toucan2/context-trace [[\"resolve connection\" {:toucan2.connection/connectable metabase.db.connection.ApplicationDB}] [\"resolve connection\" {:toucan2.connection/connectable :default}] [\"resolve connection\" {:toucan2.connection/connectable nil}] [\"with resolved query\" {:toucan2.pipeline/resolved-query {}}] [\"with parsed args\" {:toucan2.pipeline/query-type :toucan.query-type/update.update-count, :toucan2.pipeline/parsed-args {:kv-args {:id [:in (119)]}, :changes {:last_viewed_at [:case [:= :id 119] [:greatest [:coalesce :last_viewed_at #t \"0000-01-01T00:00Z\"] #t \"2025-05-30T05:10:42.828695644Z\"]]}, :queryable {}}}] [\"with model\" {:toucan2.pipeline/model :model/Dashboard}] [\"with unparsed args\" {:toucan2.pipeline/query-type :toucan.query-type/update.update-count, :toucan2.pipeline/unparsed-args (:model/Dashboard :id [:in (119)] {:last_viewed_at [:case [:= :id 119] [:greatest [:coalesce :last_viewed_at #t \"0000-01-01T00:00Z\"] #t \"2025-05-30T05:10:42.828695644Z\"]]})}]]}\n\tat org.postgresql.jdbc.PgConnection.checkClosed(PgConnection.java:1009)\n\tat org.postgresql.jdbc.PgConnection.setAutoCommit(PgConnection.java:938)\n\tat com.mchange.v2.c3p0.impl.NewProxyConnection.setAutoCommit(NewProxyConnection.java:1401)\n\tat metabase.db.connection$do_transaction.invokeStatic(connection.clj:161)\n\tat metabase.db.connection$do_transaction.invoke(connection.clj:136)\n\tat metabase.db.connection$do_with_transaction_primary_method_java_sql_Connection.invokeStatic(connection.clj:196)\n\tat metabase.db.connection$do_with_transaction_primary_method_java_sql_Connection.invoke(connection.clj:169)\n\tat methodical.impl.combo.common$partial_STAR_$fn__19889.invoke(common.clj:13)\n\tat methodical.util.FnWithMeta.invoke(util.clj:46)\n\tat toucan2.connection$do_with_transaction_around_method_toucan2_connection_default.invokeStatic(connection.clj:249)\n\tat toucan2.connection$do_with_transaction_around_method_toucan2_connection_default.invoke(connection.clj:245)\n\tat methodical.impl.combo.common$partial_STAR_$fn__19889.invoke(common.clj:13)\n\tat methodical.util.FnWithMeta.invoke(util.clj:46)\n\tat methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:58)\n\tat methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)\n\tat methodical.impl.standard.StandardMultiFn.invoke(standard.clj:212)\n\tat toucan2.tools.before_update$transduce_query_primary_method_toucan_query_type_update__STAR__toucan2_tools_before_update_before_update_default$with_connection_STAR___27861.invoke(before_update.clj:114)\n\tat toucan2.connection$bind_current_connectable_fn$fn__23214.invoke(connection.clj:104)\n\tat toucan2.connection$bind_current_connectable_fn$fn__23214.invoke(connection.clj:104)\n\tat toucan2.connection$bind_current_connectable_fn$fn__23214.invoke(connection.clj:104)\n\tat toucan2.jdbc.connection$do_with_connection_primary_method_javax_sql_DataSource.invokeStatic(connection.clj:18)\n\tat toucan2.jdbc.connection$do_with_connection_primary_method_javax_sql_DataSource.invoke(connection.clj:15)\n\tat methodical.impl.combo.common$partial_STAR_$fn__19889.invoke(common.clj:12)\n\tat methodical.util.FnWithMeta.invoke(util.clj:46)\n\tat toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)\n\tat toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)\n\tat methodical.impl.combo.common$partial_STAR_$fn__19889.invoke(common.clj:12)\n\tat methodical.util.FnWithMeta.invoke(util.clj:46)\n\tat methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)\n\tat methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)\n\tat methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)\n\tat metabase.db.connection$do_with_connection_primary_method_default.invokeStatic(connection.clj:132)\n\tat metabase.db.connection$do_with_connection_primary_method_default.invoke(connection.clj:130)\n\tat methodical.impl.combo.common$partial_STAR_$fn__19889.invoke(common.clj:12)\n\tat methodical.util.FnWithMeta.invoke(util.clj:46)\n\tat toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)\n\tat toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)\n\tat methodical.impl.combo.common$partial_STAR_$fn__19889.invoke(common.clj:12)\n\tat methodical.util.FnWithMeta.invoke(util.clj:46)\n\tat methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)\n\tat methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)\n\tat methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)\n\tat toucan2.connection$do_with_connection_primary_method_.invokeStatic(connection.clj:204)\n\tat toucan2.connection$do_with_connection_primary_method_.invoke(connection.clj:194)\n\tat methodical.impl.combo.common$partial_STAR_$fn__19889.invoke(common.clj:12)\n\tat methodical.util.FnWithMeta.invoke(util.clj:46)\n\tat toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)\n\tat toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)\n\tat methodical.impl.combo.common$partial_STAR_$fn__19889.invoke(common.clj:12)\n\tat methodical.util.FnWithMeta.invoke(util.clj:46)\n\tat methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)\n\tat methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)\n\tat methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)\n\tat toucan2.tools.before_update$transduce_query_primary_method_toucan_query_type_update__STAR__toucan2_tools_before_update_before_update_default.invokeStatic(before_update.clj:114)\n\tat toucan2.tools.before_update$transduce_query_primary_method_toucan_query_type_update__STAR__toucan2_tools_before_update_before_update_default.invoke(before_update.clj:96)\n\tat methodical.impl.combo.common$partial_STAR_$fn__19889.invoke(common.clj:15)\n\tat methodical.impl.combo.threaded$combine_methods_thread_last$fn__20199$combined_method_thread_last__20200.invoke(threaded.clj:64)\n\tat methodical.util.FnWithMeta.invoke(util.clj:46)\n\tat methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:65)\n\tat methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)\n\tat methodical.impl.standard.StandardMultiFn.invoke(standard.clj:216)\n\tat toucan2.pipeline$transduce_query_STAR_.invokeStatic(pipeline.clj:278)\n\tat toucan2.pipeline$transduce_query_STAR_.invoke(pipeline.clj:274)\n\tat toucan2.pipeline$transduce_with_model.invokeStatic(pipeline.clj:293)\n\tat toucan2.pipeline$transduce_with_model.invoke(pipeline.clj:280)\n\tat toucan2.pipeline$transduce_parsed.invokeStatic(pipeline.clj:309)\n\tat toucan2.pipeline$transduce_parsed.invoke(pipeline.clj:295)\n\tat toucan2.pipeline$transduce_unparsed.invokeStatic(pipeline.clj:317)\n\tat toucan2.pipeline$transduce_unparsed.invoke(pipeline.clj:311)\n\tat toucan2.pipeline$transduce_unparsed_with_default_rf.invokeStatic(pipeline.clj:383)\n\tat toucan2.pipeline$transduce_unparsed_with_default_rf.invoke(pipeline.clj:377)\n\tat toucan2.update$update_BANG_.invokeStatic(update.clj:53)\n\tat toucan2.update$update_BANG_.doInvoke(update.clj:50)\n\tat clojure.lang.RestFn.invoke(RestFn.java:460)\n\tat metabase.events.view_log$update_dashboard_last_viewed_at_BANG__STAR_.invokeStatic(view_log.clj:111)\n\tat metabase.events.view_log$update_dashboard_last_viewed_at_BANG__STAR_.invoke(view_log.clj:107)\n\tat grouper.core$body_fn$fn__66402.invoke(core.clj:72)\n\tat grouper.core$start_BANG_$thread__66445.invoke(core.clj:135)\n\tat clojure.lang.AFn.run(AFn.java:22)\n\tat java.base/java.lang.Thread.run(Unknown Source)\nCaused by: org.postgresql.util.PSQLException: This connection has been closed.\n\t... 80 more\n"}}
{"lvl":"ERROR","lgr":"metabase.events.view-log","m":"Failed to increment view counts","exception":{"exception_class":"clojure.lang.ExceptionInfo","exception_message":"This connection has been closed.","stacktrace":"clojure.lang.ExceptionInfo: This connection has been closed. {:toucan2/context-trace [[\"resolve connection\" {:toucan2.connection/connectable metabase.db.connection.ApplicationDB}] [\"resolve connection\" {:toucan2.connection/connectable :default}] [\"resolve connection\" {:toucan2.connection/connectable nil}]]}\n\tat org.postgresql.jdbc.PgConnection.checkClosed(PgConnection.java:1009)\n\tat org.postgresql.jdbc.PgConnection.setAutoCommit(PgConnection.java:938)\n\tat com.mchange.v2.c3p0.impl.NewProxyConnection.setAutoCommit(NewProxyConnection.java:1401)\n\tat metabase.db.connection$do_transaction.invokeStatic(connection.clj:161)\n\tat metabase.db.connection$do_transaction.invoke(connection.clj:136)\n\tat metabase.db.connection$do_with_transaction_primary_method_java_sql_Connection.invokeStatic(connection.clj:196)\n\tat metabase.db.connection$do_with_transaction_primary_method_java_sql_Connection.invoke(connection.clj:169)\n\tat methodical.impl.combo.common$partial_STAR_$fn__19889.invoke(common.clj:13)\n\tat methodical.util.FnWithMeta.invoke(util.clj:46)\n\tat toucan2.connection$do_with_transaction_around_method_toucan2_connection_default.invokeStatic(connection.clj:249)\n\tat toucan2.connection$do_with_transaction_around_method_toucan2_connection_default.invoke(connection.clj:245)\n\tat methodical.impl.combo.common$partial_STAR_$fn__19889.invoke(common.clj:13)\n\tat methodical.util.FnWithMeta.invoke(util.clj:46)\n\tat methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:58)\n\tat methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)\n\tat methodical.impl.standard.StandardMultiFn.invoke(standard.clj:212)\n\tat metabase.util.cluster_lock$do_with_cluster_lock_STAR_$with_connection_STAR___76195.invoke(cluster_lock.clj:47)\n\tat toucan2.connection$bind_current_connectable_fn$fn__23214.invoke(connection.clj:104)\n\tat toucan2.connection$bind_current_connectable_fn$fn__23214.invoke(connection.clj:104)\n\tat toucan2.connection$bind_current_connectable_fn$fn__23214.invoke(connection.clj:104)\n\tat toucan2.jdbc.connection$do_with_connection_primary_method_javax_sql_DataSource.invokeStatic(connection.clj:18)\n\tat toucan2.jdbc.connection$do_with_connection_primary_method_javax_sql_DataSource.invoke(connection.clj:15)\n\tat methodical.impl.combo.common$partial_STAR_$fn__19889.invoke(common.clj:12)\n\tat methodical.util.FnWithMeta.invoke(util.clj:46)\n\tat toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)\n\tat toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)\n\tat methodical.impl.combo.common$partial_STAR_$fn__19889.invoke(common.clj:12)\n\tat methodical.util.FnWithMeta.invoke(util.clj:46)\n\tat methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)\n\tat methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)\n\tat methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)\n\tat metabase.db.connection$do_with_connection_primary_method_default.invokeStatic(connection.clj:132)\n\tat metabase.db.connection$do_with_connection_primary_method_default.invoke(connection.clj:130)\n\tat methodical.impl.combo.common$partial_STAR_$fn__19889.invoke(common.clj:12)\n\tat methodical.util.FnWithMeta.invoke(util.clj:46)\n\tat toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)\n\tat toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)\n\tat methodical.impl.combo.common$partial_STAR_$fn__19889.invoke(common.clj:12)\n\tat methodical.util.FnWithMeta.invoke(util.clj:46)\n\tat methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)\n\tat methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)\n\tat methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)\n\tat toucan2.connection$do_with_connection_primary_method_.invokeStatic(connection.clj:204)\n\tat toucan2.connection$do_with_connection_primary_method_.invoke(connection.clj:194)\n\tat methodical.impl.combo.common$partial_STAR_$fn__19889.invoke(common.clj:12)\n\tat methodical.util.FnWithMeta.invoke(util.clj:46)\n\tat toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)\n\tat toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)\n\tat methodical.impl.combo.common$partial_STAR_$fn__19889.invoke(common.clj:12)\n\tat methodical.util.FnWithMeta.invoke(util.clj:46)\n\tat methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)\n\tat methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)\n\tat methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)\n\tat metabase.util.cluster_lock$do_with_cluster_lock_STAR_.invokeStatic(cluster_lock.clj:47)\n\tat metabase.util.cluster_lock$do_with_cluster_lock_STAR_.invoke(cluster_lock.clj:45)\n\tat metabase.util.cluster_lock$do_with_cluster_lock76202__76203$do_with_cluster_lock_STAR__STAR___76205.invoke(cluster_lock.clj:71)\n\tat clojure.lang.AFn.applyToHelper(AFn.java:152)\n\tat clojure.lang.AFn.applyTo(AFn.java:144)\n\tat clojure.core$apply.invokeStatic(core.clj:667)\n\tat clojure.core$apply.invoke(core.clj:662)\n\tat metabase.util.retry$decorate$fn$reify__75731.call(retry.clj:101)\n\tat io.github.resilience4j.retry.Retry.lambda$decorateCallable$5(Retry.java:306)\n\tat metabase.util.retry$decorate$fn__75730.doInvoke(retry.clj:102)\n\tat clojure.lang.RestFn.invoke(RestFn.java:400)\n\tat metabase.util.retry$make75735__75736$fn__75737.invoke(retry.clj:111)\n\tat metabase.util.cluster_lock$do_with_cluster_lock76202__76203.invokeStatic(cluster_lock.clj:75)\n\tat metabase.util.cluster_lock$do_with_cluster_lock76202__76203.invoke(cluster_lock.clj:56)\n\tat metabase.util.cluster_lock$do_with_cluster_lock76202__76203.invokeStatic(cluster_lock.clj:68)\n\tat metabase.util.cluster_lock$do_with_cluster_lock76202__76203.invoke(cluster_lock.clj:56)\n\tat metabase.events.view_log$increment_view_counts_BANG__STAR_.invokeStatic(view_log.clj:44)\n\tat metabase.events.view_log$increment_view_counts_BANG__STAR_.invoke(view_log.clj:29)\n\tat grouper.core$body_fn$fn__66402.invoke(core.clj:72)\n\tat grouper.core$start_BANG_$thread__66445.invoke(core.clj:135)\n\tat clojure.lang.AFn.run(AFn.java:22)\n\tat java.base/java.lang.Thread.run(Unknown Source)\nCaused by: org.postgresql.util.PSQLException: This connection has been closed.\n\t... 75 more\n"}}
```

### Information about your Metabase installation

```JSON
v1.54.10-X01
```

### Severity

P2

### Additional context

Only happens on certain dashboards, usually after adding one specific question (id 470, also when duplicating it, like id 654) and linking dashboard filters to it, but unsure if this is related.

Context on slack [here](https://metaboat.slack.com/archives/CKZEMT1MJ/p1748869553047389?thread_ts=1748532253.613829&cid=CKZEMT1MJ)

--- a/src/metabase/dashboards/models/dashboard.clj
+++ b/src/metabase/dashboards/models/dashboard.clj
@@ -16,8 +16,6 @@
    [metabase.parameters.schema :as parameters.schema]
    [metabase.permissions.core :as perms]
    [metabase.public-sharing.core :as public-sharing]
-   ^{:clj-kondo/ignore [:deprecated-namespace]}
-   [metabase.pulse.core :as pulse]
    [metabase.queries.core :as queries]
    [metabase.query-processor.metadata :as qp.metadata]
    [metabase.search.core :as search]
@@ -100,11 +98,6 @@
       (collection/check-collection-namespace :model/Dashboard (:collection_id dashboard))
       (when (:archived changes)
         (t2/delete! :model/Pulse :dashboard_id (u/the-id dashboard))))))
-
-(t2/define-after-update :model/Dashboard
-  [dashboard]
-  ; TODO -- should this be done on `:event/dashboard-update` ?
-  (pulse/update-dashboard-subscription-pulses! dashboard))
 
 (defn- migrate-parameter [p]
   (cond-> p

--- a/src/metabase/models/interface.clj
+++ b/src/metabase/models/interface.clj
@@ -501,7 +501,7 @@
         ; don't stomp on `:updated_at` if it's already explicitly specified.
         changes-already-include-updated-at? (some #{:updated_at} changed-fields)
         has-non-ignored-fields? (seq (set/difference changed-fields (non-timestamped-fields obj)))
-        should-set-updated-at? (or (empty? changed-fields) (and has-non-ignored-fields? (not changes-already-include-updated-at?)))]
+        should-set-updated-at? (and has-non-ignored-fields? (not changes-already-include-updated-at?))]
     (cond-> obj
       should-set-updated-at? (assoc :updated_at (now)))))
 

--- a/src/metabase/native_query_snippets/models/native_query_snippet.clj
+++ b/src/metabase/native_query_snippets/models/native_query_snippet.clj
@@ -28,12 +28,11 @@
     (collection/check-collection-namespace :model/NativeQuerySnippet (:collection_id snippet))))
 
 (t2/define-before-update :model/NativeQuerySnippet
-  [{:keys [id], :as snippet}]
-  (u/prog1 (t2/changes snippet)
+  [snippet]
+  (u/prog1 snippet
     ;; throw an Exception if someone tries to update creator_id
-    (when (contains? <> :creator_id)
-      (when (not= (:creator_id <>) (t2/select-one-fn :creator_id :model/NativeQuerySnippet :id id))
-        (throw (UnsupportedOperationException. (tru "You cannot update the creator_id of a NativeQuerySnippet.")))))
+    (when (contains? (t2/changes <>) :creator_id)
+      (throw (UnsupportedOperationException. (tru "You cannot update the creator_id of a NativeQuerySnippet."))))
     (collection/check-collection-namespace :model/NativeQuerySnippet (:collection_id snippet))))
 
 (defmethod serdes/hash-fields :model/NativeQuerySnippet

--- a/src/metabase/pulse/core.clj
+++ b/src/metabase/pulse/core.clj
@@ -3,19 +3,15 @@
 
   This namespace is deprecated, soon everything will be migrated to notifications."
   (:require
-   [metabase.pulse.dashboard-subscription]
    [metabase.pulse.models.pulse]
    [metabase.pulse.update-alerts]
    [potemkin :as p]))
 
 (comment
-  metabase.pulse.dashboard-subscription/keep-me
   metabase.pulse.models.pulse/keep-me
   metabase.pulse.update-alerts/keep-me)
 
 (p/import-vars
- [metabase.pulse.dashboard-subscription
-  update-dashboard-subscription-pulses!]
  [metabase.pulse.models.pulse
   card->ref
   create-pulse!

--- a/src/metabase/pulse/events/dashboard_subscription.clj
+++ b/src/metabase/pulse/events/dashboard_subscription.clj
@@ -1,21 +1,25 @@
-(ns metabase.pulse.dashboard-subscription
+(ns metabase.pulse.events.dashboard-subscription
   (:require
    [clojure.set :as set]
    [metabase.app-db.core :as app-db]
+   [metabase.events.core :as events]
    [metabase.pulse.models.pulse :as models.pulse]
    [metabase.pulse.models.pulse-card :as pulse-card]
    [metabase.util :as u]
+   [methodical.core :as m]
    [toucan2.core :as t2]))
 
-;; TODO -- should this be done on `:event/dashboard-update` ??
-(defn update-dashboard-subscription-pulses!
+(derive ::dashboard-update :metabase/event)
+(derive :event/dashboard-update ::dashboard-update)
+
+(m/defmethod events/publish-event! ::dashboard-update
   "Updates the pulses' names and collection IDs, and syncs the PulseCards"
-  [dashboard]
+  [_ {dashboard :object}]
   (let [dashboard-id (u/the-id dashboard)
         affected     (app-db/query
                       {:select-distinct [[:p.id :pulse-id] [:pc.card_id :card-id]]
                        :from            [[:pulse :p]]
-                       :join            [[:pulse_card :pc] [:= :p.id :pc.pulse_id]]
+                       :left-join       [[:pulse_card :pc] [:= :p.id :pc.pulse_id]]
                        :where           [:= :p.dashboard_id dashboard-id]})]
     (when-let [pulse-ids (seq (distinct (map :pulse-id affected)))]
       (let [correct-card-ids     (->> (app-db/query

--- a/src/metabase/pulse/init.clj
+++ b/src/metabase/pulse/init.clj
@@ -1,5 +1,6 @@
 (ns metabase.pulse.init
   (:require
+   [metabase.pulse.events.dashboard-subscription]
    [metabase.pulse.events.report-timezone-updated]
    [metabase.pulse.task.email-remove-legacy-pulse]
    [metabase.pulse.task.send-pulses]))

--- a/src/metabase/pulse/models/pulse.clj
+++ b/src/metabase/pulse/models/pulse.clj
@@ -95,7 +95,7 @@
                (contains? notification :dashboard_id)
                (not= (:dashboard_id notification) dashboard_id))
       (throw (ex-info (tru "dashboard ID of a dashboard subscription cannot be modified") notification))))
-  (u/prog1 (t2/changes notification)
+  (u/prog1 notification
     (assert-valid-parameters notification)
     (collection/check-collection-namespace :model/Pulse (:collection_id notification))))
 

--- a/src/metabase/queries/models/parameter_card.clj
+++ b/src/metabase/queries/models/parameter_card.clj
@@ -35,8 +35,8 @@
 
 (t2/define-before-update :model/ParameterCard
   [pc]
-  (u/prog1 (t2/changes pc)
-    (when (:parameterized_object_type <>)
+  (u/prog1 pc
+    (when (:parameterized_object_type (t2/changes <>))
       (validate-parameterized-object-type <>))))
 
 (defn delete-all-for-parameterized-object!

--- a/src/metabase/segments/models/segment.clj
+++ b/src/metabase/segments/models/segment.clj
@@ -80,12 +80,11 @@
   ([model pk]
    (mi/can-read? (t2/select-one model pk))))
 
-(t2/define-before-update :model/Segment  [{:keys [id], :as segment}]
-  (u/prog1 (t2/changes segment)
-    ;; throw an Exception if someone tries to update creator_id
-    (when (contains? <> :creator_id)
-      (when (not= (:creator_id <>) (t2/select-one-fn :creator_id :model/Segment :id id))
-        (throw (UnsupportedOperationException. (tru "You cannot update the creator_id of a Segment.")))))))
+(t2/define-before-update :model/Segment  [segment]
+  ;; throw an Exception if someone tries to update creator_id
+  (when (contains? (t2/changes segment) :creator_id)
+    (throw (UnsupportedOperationException. (tru "You cannot update the creator_id of a Segment."))))
+  segment)
 
 (defmethod mi/perms-objects-set :model/Segment
   [segment read-or-write]

--- a/src/metabase/view_log/events/view_log.clj
+++ b/src/metabase/view_log/events/view_log.clj
@@ -16,6 +16,18 @@
    [steffan-westcott.clj-otel.api.trace.span :as span]
    [toucan2.core :as t2]))
 
+(def ^:private dashboard-statistics-lock
+  "keyword to use for locking dashboard updates that can deadlock"
+  ::dashboard-statistics-lock)
+
+(defn- view-count-lock
+  [model]
+  (case model
+    :model/Card cluster-lock/card-statistics-lock
+    :model/Dashboard dashboard-statistics-lock
+    (keyword "metabase.events.view_log"
+             (str (name (t2/table-name model)) "-view-count"))))
+
 (defn- group-by-frequency
   "Given a list of items, returns a map of frequencies to items.
     (group-by-frequency [:a :a :b :b :c :c :c])
@@ -36,10 +48,7 @@
                              items)]
       (doseq [[model ids] model->ids]
         (let [cnt->ids (group-by-frequency ids)
-              lock-name (if (= model :model/Card)
-                          cluster-lock/card-statistics-lock ;; need to use a shared lock for all updates to the card table
-                          (keyword "metabase.events.view_log"
-                                   (str (name (t2/table-name model)) "-view-count")))]
+              lock-name (view-count-lock model)]
           (log/debugf "Writing %d items to %s view counts with lock %s" (count ids) model lock-name)
           (cluster-lock/with-cluster-lock lock-name
             (t2/query {:update (t2/table-name model)
@@ -108,11 +117,13 @@
   (let [dashboard-id->timestamp (update-vals (group-by :id dashboard-id-timestamps)
                                              (fn [xs] (apply t/max (map :timestamp xs))))]
     (try
-      (t2/update! :model/Dashboard :id [:in (keys dashboard-id->timestamp)]
-                  {:last_viewed_at (into [:case]
-                                         (mapcat (fn [[id timestamp]]
-                                                   [[:= :id id] [:greatest [:coalesce :last_viewed_at (t/offset-date-time 0)] timestamp]])
-                                                 dashboard-id->timestamp))})
+      (cluster-lock/with-cluster-lock dashboard-statistics-lock
+        (t2/update! :model/Dashboard :id [:in (keys dashboard-id->timestamp)]
+                    {:last_viewed_at (into [:case]
+                                           (mapcat (fn [[id timestamp]]
+                                                     [[:= :id id] [:greatest [:coalesce :last_viewed_at (t/offset-date-time 0)] timestamp]])
+                                                   dashboard-id->timestamp))
+                     :updated_at :updated_at})) ;; setting last_viewed_at should not update the updated_at column
       (catch Exception e
         (log/error e "Failed to update dashboard last_viewed_at")))))
 

--- a/src/metabase/warehouse_schema/api/table.clj
+++ b/src/metabase/warehouse_schema/api/table.clj
@@ -98,7 +98,7 @@
   (when-let [changes (not-empty (u/select-keys-when body
                                                     :non-nil [:display_name :show_in_getting_started :entity_type :field_order]
                                                     :present [:description :caveats :points_of_interest :visibility_type]))]
-    (api/check-500 (pos? (t2/update! :model/Table id changes))))
+    (t2/update! :model/Table id changes))
   (let [updated-table        (t2/select-one :model/Table :id id)
         changed-field-order? (not= (:field_order updated-table) (:field_order existing-table))]
     (if changed-field-order?

--- a/src/metabase/warehouse_schema/models/field.clj
+++ b/src/metabase/warehouse_schema/models/field.clj
@@ -133,8 +133,8 @@
 
 (t2/define-before-update :model/Field
   [field]
-  (u/prog1 (t2/changes field)
-    (when (false? (:active <>))
+  (u/prog1 field
+    (when (false? (:active (t2/changes <>)))
       (t2/update! :model/Field {:fk_target_field_id (:id field)} {:semantic_type      nil
                                                                   :fk_target_field_id nil}))))
 

--- a/test/metabase/channel/models/channel_test.clj
+++ b/test/metabase/channel/models/channel_test.clj
@@ -23,7 +23,7 @@
                                          :enabled true}]
     (testing "do not try to delete pulse-channel if active doesn't change"
       (is (pos? (t2/update! :model/Channel id {:name "New name"})))
-      (is (pos? (t2/update! :model/Channel id {:active true})))
+      (is (zero? (t2/update! :model/Channel id {:active true})))
       (is (t2/exists? :model/PulseChannel pc-id)))
 
     (testing "deactivate channel"

--- a/test/metabase/dashboards/api_test.clj
+++ b/test/metabase/dashboards/api_test.clj
@@ -5058,3 +5058,37 @@
                                     :base_type :type/Text}]}
                 (:param_fields (mt/with-test-user :crowberto
                                  (#'api.dashboard/get-dashboard (:id dashboard))))))))))
+
+(deftest post-update-test
+  (mt/with-temp [:model/Collection    {collection-id-1 :id} {}
+                 :model/Collection    {collection-id-2 :id} {}
+                 :model/Dashboard     {dashboard-id :id}    {:name "Lucky the Pigeon's Lucky Stuff", :collection_id collection-id-1}
+                 :model/Card          {card-id :id}         {}
+                 :model/Pulse         {pulse-id :id}        {:dashboard_id dashboard-id, :collection_id collection-id-1}
+                 :model/DashboardCard {dashcard-id :id}     {:dashboard_id dashboard-id, :card_id card-id}
+                 :model/PulseCard     _                     {:pulse_id pulse-id, :card_id card-id, :dashboard_card_id dashcard-id}]
+    (testing "Pulse name and collection-id updates"
+      (mt/user-http-request :crowberto :put 200 (str "dashboard/" dashboard-id)
+                            {:name "Lucky's Close Shaves" :collection_id collection-id-2})
+      (is (= "Lucky's Close Shaves"
+             (t2/select-one-fn :name :model/Pulse :id pulse-id)))
+      (is (= collection-id-2
+             (t2/select-one-fn :collection_id :model/Pulse :id pulse-id))))))
+
+(deftest post-update-card-sync-test
+  (mt/with-temp [:model/Collection    {collection-id-1 :id} {}
+                 :model/Dashboard     {dashboard-id :id}    {:name "Lucky the Pigeon's Lucky Stuff", :collection_id collection-id-1}
+                 :model/Card          {card-id :id}         {}
+                 :model/Card          {new-card-id :id}     {}
+                 :model/Pulse         {pulse-id :id}        {:dashboard_id dashboard-id, :collection_id collection-id-1}
+                 :model/DashboardCard {dashcard-id :id}     {:dashboard_id dashboard-id, :card_id card-id}
+                 :model/PulseCard     _                     {:pulse_id pulse-id, :card_id card-id, :dashboard_card_id dashcard-id}]
+    (testing "PulseCard syncing"
+      (mt/user-http-request :crowberto :put 200 (str "dashboard/" dashboard-id)
+                            {:dashcards [{:id 100
+                                          :card_id new-card-id
+                                          :row     0
+                                          :col     0
+                                          :size_x  4
+                                          :size_y  4}]})
+      (is (not (nil? (t2/select-one :model/PulseCard :card_id new-card-id)))))))

--- a/test/metabase/dashboards/models/dashboard_test.clj
+++ b/test/metabase/dashboards/models/dashboard_test.clj
@@ -99,30 +99,6 @@
         (is (nil? (t2/select-one :model/Pulse pulse-id)))
         (is (= 0 (count (pulse-channel-test/send-pulse-triggers pulse-id))))))))
 
-(deftest post-update-test
-  (mt/with-temp [:model/Collection    {collection-id-1 :id} {}
-                 :model/Collection    {collection-id-2 :id} {}
-                 :model/Dashboard     {dashboard-id :id}    {:name "Lucky the Pigeon's Lucky Stuff", :collection_id collection-id-1}
-                 :model/Card          {card-id :id}         {}
-                 :model/Pulse         {pulse-id :id}        {:dashboard_id dashboard-id, :collection_id collection-id-1}
-                 :model/DashboardCard {dashcard-id :id}     {:dashboard_id dashboard-id, :card_id card-id}
-                 :model/PulseCard     _                     {:pulse_id pulse-id, :card_id card-id, :dashboard_card_id dashcard-id}]
-    (testing "Pulse name and collection-id updates"
-      (t2/update! :model/Dashboard dashboard-id {:name "Lucky's Close Shaves" :collection_id collection-id-2})
-      (is (= "Lucky's Close Shaves"
-             (t2/select-one-fn :name :model/Pulse :id pulse-id)))
-      (is (= collection-id-2
-             (t2/select-one-fn :collection_id :model/Pulse :id pulse-id))))
-    (testing "PulseCard syncing"
-      (mt/with-temp [:model/Card {new-card-id :id}]
-        (dashboard/add-dashcards! dashboard-id [{:card_id new-card-id
-                                                 :row     0
-                                                 :col     0
-                                                 :size_x  4
-                                                 :size_y  4}])
-        (t2/update! :model/Dashboard dashboard-id {:name "Lucky's Close Shaves"})
-        (is (not (nil? (t2/select-one :model/PulseCard :card_id new-card-id))))))))
-
 (deftest parameter-card-test
   (testing "A new dashboard creates a new ParameterCard"
     (mt/with-temp [:model/Card      {card-id :id}      {}

--- a/test/metabase/segments/models/segment_test.clj
+++ b/test/metabase/segments/models/segment_test.clj
@@ -44,7 +44,7 @@
              (t2/update! :model/Segment id {:creator_id nil}))))
 
       (testing "calling `update!` with a value that is the same as the current value shouldn't throw an Exception"
-        (is (= 1
+        (is (= 0
                (t2/update! :model/Segment id {:creator_id (mt/user->id :rasta)})))))))
 
 (deftest identity-hash-test

--- a/test/metabase/sync/sync_metadata/fields_test.clj
+++ b/test/metabase/sync/sync_metadata/fields_test.clj
@@ -351,7 +351,9 @@
               (t2/with-transaction [_ t2.connection/*current-connectable* {:rollback-only true}]
                 ;; 1. delete the fields that were just synced
                 (t2/delete! :model/Field :table_id [:in (map :id tables)])
-                ;; 2. sync the metadata for each table
+                ;; 2. reset the sync status for each table
+                (t2/update! :model/Table :id [:in (map :id tables)] {:initial_sync_status "incomplete"})
+                ;; 3. sync the metadata for each table
                 (if (= "for entire DB" message)
                   (let [tables-updated (atom nil)
                         original-set-initial-table-sync-complete-for-db! sync-util/set-initial-table-sync-complete-for-db!]

--- a/test/metabase/warehouse_schema/api/table_test.clj
+++ b/test/metabase/warehouse_schema/api/table_test.clj
@@ -1140,19 +1140,19 @@
 (deftest field-ordering-test
   (let [original-field-order (t2/select-one-fn :field_order :model/Table :id (mt/id :venues))]
     (try
-      (testing "Cane we set alphabetical field ordering?"
+      (testing "Can we set alphabetical field ordering?"
         (is (= ["CATEGORY_ID" "ID" "LATITUDE" "LONGITUDE" "NAME" "PRICE"]
                (->> (mt/user-http-request :crowberto :put 200 (format "table/%s" (mt/id :venues))
                                           {:field_order :alphabetical})
                     :fields
                     (map :name)))))
-      (testing "Cane we set smart field ordering?"
+      (testing "Can we set smart field ordering?"
         (is (= ["ID" "NAME" "CATEGORY_ID" "LATITUDE" "LONGITUDE" "PRICE"]
                (->> (mt/user-http-request :crowberto :put 200 (format "table/%s" (mt/id :venues))
                                           {:field_order :smart})
                     :fields
                     (map :name)))))
-      (testing "Cane we set database field ordering?"
+      (testing "Can we set database field ordering?"
         (is (= ["ID" "NAME" "CATEGORY_ID" "LATITUDE" "LONGITUDE" "PRICE"]
                (->> (mt/user-http-request :crowberto :put 200 (format "table/%s" (mt/id :venues))
                                           {:field_order :database})

